sailfishusermanager.cpp

```diff
@@ -535,3 +536,53 @@ void SailfishUserManager::updateEnvironment(uint uid)
         file.close();
     }
 }
+
+QStringList SailfishUserManager::usersGroups(uint uid)
+{
+    m_timer->start();
+    return m_lu->groups(uid);
+}
+
+void SailfishUserManager::usersGroupsAdd(uint uid, const QString& group)
+{
+    m_timer->start();
+    if (!checkAccessRights(uid))
+        return;
+
+    struct passwd *pwd = getpwuid(uid);
+    if (!pwd) {
+        auto message = QStringLiteral("User not found");
+        qCWarning(lcSUM) << message;
+        sendErrorReply(QDBusError::InvalidArgs, message);
+        return;
+    }
+
+    if (!m_lu->addUserToGroup(pwd->pw_name, group)) {
+        auto message = QStringLiteral("Failed to add user to group");
+        qCWarning(lcSUM) << message;
+        sendErrorReply(QDBusError::InvalidArgs, message);
+        return;
+    }
+}
+
+void SailfishUserManager::usersGroupsRemove(uint uid, const QString& group)
+{
+    m_timer->start();
+    if (!checkAccessRights(uid))
+        return;
+
+    struct passwd *pwd = getpwuid(uid);
+    if (!pwd) {
+        auto message = QStringLiteral("User not found");
+        qCWarning(lcSUM) << message;
+        sendErrorReply(QDBusError::InvalidArgs, message);
```

Comment:
dedicated error message identifier strings similar to other strings. E.g. this could be SailfishUserManagerErrorUserNotFound

SailfishUserManagerErrorHomeCreateFailed...

Reply:
Added errors